### PR TITLE
5208 gaps item variant service layer

### LINKED ISSUE #5208: GAPS: item_variant server & graphql framework
## Describe the solution you'd like 🎁

Able to query item_variant, and packaging_variant from graphql (doesn't have to be final API)
Insert/update won't be in this PR...

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #5208

# 👩🏻‍💻 What does this PR do?

Service layer with upsert and delete for, `item_variant` and `packaging_variant`
Will do a new PR for graphql based on https://github.com/msupply-foundation/open-msupply/pull/5227

## 💌 Any notes for the reviewer?

Haven't done much if any validation.

Will need to add this to next PR. Added some todos in the test cases for validation. Interested in feedback on what validation is needed?

# 🧪 Testing

- [ ]  No QA needed, it will be tested as part of the UI stuff in the next PR
- [ ] Check if code test cases make sense, and what needs to be added...

# 📃 Documentation

- [X] **Part of an epic**: documentation will be completed for the feature as a whole

### CODE REVIEW

File: server/service/src/item/item_variant/test/mod.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    #[actix_rt::test]
+    async fn validate_item_variant() {
+        // TODO validation tests
+
+        // Test that the item variant name is set?
+
+        // Test that we can't create a record with an item_id that doesn't exist
+
+        // Test that we can't change the item_id on an existing record???
```
Comment (lache-melvin) [inline server/service/src/item/item_variant/test/mod.rs:148]: Hm, this one is a good point! I guess fine to determine if we are doing update or insert in the validation and apply some different checks accordingly! Same with the unique name check

File: server/service/src/item/packaging_variant/upsert.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        volume_per_unit,
+        deleted_datetime: None,
+    }
+}
+
+fn validate(
+    connection: &StorageConnection,
+    input: &UpsertPackagingVariant,
+) -> Result<(), UpsertPackagingVariantError> {
+    let item_variant = check_item_variant_exists(connection, &input.item_variant_id)?;
```
Comment (lache-melvin) [inline server/service/src/item/packaging_variant/upsert.rs:75]: Wondering if we can actually get away without this step - given graphql layer, you can only insert packaging variant through item variant 👀

**Review by lache-melvin — APPROVED** [review]
Awesome thanks James! Cool to see you removing the _dsl stuff too 🔥

File: server/service/src/item/packaging_variant/upsert.rs (hunk 2 above)
Comment (jmbrunskill) [reply server/service/src/item/packaging_variant/upsert.rs:75]: Yeah I guess it depends if we want to keep this option available in the service layer?

Doesn't hurt I think...

**Pull request merged**